### PR TITLE
Modified the computeMinIntrinsicHeight method to calculate the height properly

### LINKED ISSUE #75509: TextFields compute their min intrinsic height incorrectly
When creating TextField widgets, and passing a small contentPadding to the decoration, the TextField will cause an overflow. This is because the Decorator Render object compute its MinIntrinsicHeight incorrectly.

The following sample causes an overflow.

<details><summary>Code</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: TextFieldOverflowPage(),
    );
  }
}

class TextFieldOverflowPage extends StatefulWidget {
  const TextFieldOverflowPage();

  @override
  _TextFieldOverflowPageState createState()  => _TextFieldOverflowPageState();
}

class _TextFieldOverflowPageState extends State<TextFieldOverflowPage> {

  @override
  Widget build(BuildContext context) {

    return Scaffold(
      body: SafeArea(
        child: Center(
          child: Container(
            color: Colors.pink,
            child: IntrinsicHeight(
              child: Column(
                children: <Widget>[
                  TextField(
                    decoration: InputDecoration(
                      labelText: "Type something",
                      helperText: "Let's cause some overflow!",
                      contentPadding: EdgeInsets.zero,
                    ),
                  ),
                ],
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

</details>

This happens on the latest stable, and also beta (1.26) channels.

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel stable, 1.22.2, on Mac OS X 10.15.7 19H15, locale en-US)
    • Flutter version 1.22.2 at /Users/danny.valente/Library/flutter/1.22.2-stable
    • Framework revision 84f3d28555 (9 weeks ago), 2020-10-15 16:26:19 -0700
    • Engine revision b8752bbfff
    • Dart version 2.10.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/danny.valente/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
```

</details>

### PR DESCRIPTION
**Pull request by dannyvalentesonos**
Modified the computeMinIntrinsicHeight method to calculate the height  properly, by adding the containerHeight with the subtextHeight after determining the minimum containerHeight, instead of adding the subtextHeight early, and causing it to be ignored in some cases when the minContainerHeight is larger.

The problem is that the Sub Text height must still be applied after determining the ConainerHeight, which could increase based on the minimum height the container should be.

This fixes #75509

### CODE REVIEW

**Comment by dannyvalentesonos** [comment]
@justinmc I think this is in your area, but if I'm wrong, would you please point me to who might be? Thanks!

**Comment by justinmc** [comment]
Yup this is me 👍 

Looking at your explanation of the problem and your solution here, I suspect that you're right and something was wrong with the intrinsic height calculation, but I'm not 100% sure what it is or if this fixes it completely.  Could you clarify?  I've run your example code in the linked issue and played around with it.  If I simplify it to the code below, does that illustrate the problem?

<details>

<summary>Code</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: TextFieldOverflowPage(),
    );
  }
}

class TextFieldOverflowPage extends StatefulWidget {
  const TextFieldOverflowPage();

  @override
  _TextFieldOverflowPageState createState()  => _TextFieldOverflowPageState();
}

class _TextFieldOverflowPageState extends State<TextFieldOverflowPage> {

  @override
  Widget build(BuildContext context) {

    return Scaffold(
      body: SafeArea(
        child: Center(
          child: Container(
            color: Colors.pink,
            child: IntrinsicHeight(
              child: Column(
                children: <Widget>[
                  TextField(
                    decoration: InputDecoration(
                      labelText: "Type something",
                      helperText: "Let's cause some overflow!",
                      contentPadding: EdgeInsets.zero,
                    ),
                  ),
                ],
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

</details>

**Comment by dannyvalentesonos** [comment]
> Yup this is me 👍
> 
> Looking at your explanation of the problem and your solution here, I suspect that you're right and something was wrong with the intrinsic height calculation, but I'm not 100% sure what it is or if this fixes it completely. Could you clarify? I've run your example code in the linked issue and played around with it. If I simplify it to the code below, does that illustrate the problem?
> 
> Code

Thanks for responding @justinmc .

Your code does cause the problem. Thanks for a much simpler repro!

To explain, the TextField widget will always force the content container to be a minimum of `decoration.floatingLabelHeight`, and then lay out the subtext under that. However, the minimum intrinsic height calculation before my change adds the subtext height to its content container calculation, which it then takes the maximum of it, or `decoration.floatingLabelHeight`.

When the content padding is 0, the content container ends up being less than `decoration.floatingLabelHeight` so the total height should end up being `decoration.floatingLabelHeight` + the subtext height, seeing the content container should be a minimum of `decoriation.floatingLabelHeight`.

The intrinsic height of the elements that make up the content container itself is less than `decoration.floatingLabelHeight` so we need to take that value and _**then**_ add the subtext height to it.

**Comment by dannyvalentesonos** [comment]
@justinmc I wanted to get you some more info.

Here is a screenshot showing the overall height of a TextField with 0 content padding, when it is not in a IntrinsicHeight widget.

Screen Shot 2021-02-09 at 9 59 42 AM

Here is a screenshot the height of the container for the same widget (content padding still set to 0).

Screen Shot 2021-02-09 at 9 45 19 AM

Here is a screenshot of the height of the subtext for the same widget.

Screen Shot 2021-02-09 at 9 44 21 AM

The height of the sub text is 14, and the gap above it is a const 8.0 in size, which means 22.0.

So, when the padding is set to 0, it's important to note that the content container itself is set to a minimum of 48.

Breaking into the code, without my change looks like this:

Screen Shot 2021-02-09 at 9 48 11 AM

As you can see, `containerHeight` has been calculated to be 57.0 which includes the 22.0 for the `subText`, but this overall is shy of the overall 70.0 which the textField ends up being.

Now, looking at the code, with my changes:

Screen Shot 2021-02-09 at 9 48 40 AM

The containerHeight is now calculated to be 35.0, which will end up getting set to the max of the minContainerHeight (48.0) and the returned height is now 70.0 when adding the `subText` (22.0) to it.

I hope this gives you enough information. Let me know if you have any further questions. Thanks!

**Comment by dannyvalentesonos** [comment]
@justinmc @goderbauer Any chance you can take a look at this please? Thanks!

**Review by justinmc — COMMENTED** [review]
Sorry I missed your follow up comments on this.  I really appreciate the clear explanation.  I see what you mean now, and I think you're right.  I tried out your code in a couple of cases and it seems to always work, and all of the tests are passing 👍 .

The only thing this still needs is its own test.  Could you maybe write a test in input_decorator_test.dart that builds the app from my repro code above, and verify that it does not overflow?

**Comment by dannyvalentesonos** [comment]
Thanks @justinmc. I did look at writing a test, but couldn't really figure out how to verify that no overflow happens. Is there some test I can look at that does something like that so I can build off of it? Thanks

**Comment by justinmc** [comment]
Let me know if this one helps: https://github.com/flutter/flutter/blob/9fdda01252c99559af4a33b2e2feaf3d92d8fcea/packages/flutter/test/rendering/debug_overflow_indicator_test.dart#L23

**Comment by dannyvalentesonos** [comment]
Yep, that looks good. Thanks!

**Comment by dannyvalentesonos** [comment]
@justinmc I've added a test which I've verified fails when I remove the fix, and passes with the fix. Let me know what you think. Thanks.

**Comment by dannyvalentesonos** [comment]
@justinmc Pinging you just in case you missed the addition of the test. Thanks!

File: packages/flutter/test/material/input_decorator_test.dart [hunk 1]
```diff
@@ -4459,4 +4459,37 @@ void main() {
...
+              ),
+            ),
+          ),
+        ),
+      ));
+    } finally {
+      FlutterError.onError = oldHandler;
+    }
+    
+    expect(errorDetails?.toString() ?? "", isNot(contains("A RenderFlex overflowed by ")));
```
Comment (justinmc) [inline packages/flutter/test/material/input_decorator_test.dart:4493]: Nit: Is there some other error happening here besides the overflow error?  Or could you do `expect(tester.takeException(), isNull);`?

File: packages/flutter/test/material/input_decorator_test.dart [hunk 2]
```diff
@@ -4459,4 +4459,37 @@ void main() {
     expect(errorDetails?.toString(), contains("InputDecorator's children reported a negative baseline"));
     expect(errorDetails?.toString(), contains('RenderStack'));
   });
+
```
Comment (justinmc) [inline packages/flutter/test/material/input_decorator_test.dart:4462]: Add a comment here linking to the issue that this is testing:

```
// Regression test for: https://github.com/flutter/flutter/issues/75509
```

**Review by justinmc — COMMENTED** [review]
You've got a trailing space somewhere that's causing an analyzer error.  Otherwise, just some small nits below.  I this is ready to merge once it's passing the tests again.

File: packages/flutter/test/material/input_decorator_test.dart (hunk 2 above)
Comment (dannyvalentesonos) [reply packages/flutter/test/material/input_decorator_test.dart:4462]: Did you mean to put this comment on line 4493 instead?

File: packages/flutter/test/material/input_decorator_test.dart (hunk 1 above)
Comment (dannyvalentesonos) [reply packages/flutter/test/material/input_decorator_test.dart:4493]: I was thinking of this as well, but wasn't sure what was the preferred method. I'll update. 
Thanks.

File: packages/flutter/test/material/input_decorator_test.dart (hunk 2 above)
Comment (justinmc) [reply packages/flutter/test/material/input_decorator_test.dart:4462]: Oh sorry you got it right, it should be inside the `testWidgets` block 👍

**Comment by dannyvalentesonos** [comment]
> You've got a trailing space somewhere that's causing an analyzer error. Otherwise, just some small nits below. I this is ready to merge once it's passing the tests again.

Is there anything else wrong. I'm not sure why flutter-build is failing or if anything else needs to be changed. Thanks!

**Comment by dannyvalentesonos** [comment]
@justinmc Is there anything with my changes causing flutter-build to fail? When I look at the details for flutter-build, I never understand how to digest that dashboard.
Thanks!

**Comment by justinmc** [comment]
The flutter-build check is just checking whether the current master branch is passing all tests.  So it is not affected by this PR since this PR hasn't been merged yet.  In general if it's red, you can just wait for it to turn green and then merge, or you can add the "waiting for tree to go green" label and the PR will be merged automatically when it's green.

It looks like framework_tests-misc-linux is also failing, but it doesn't seem related.  Can you try merging in master and see if that fixes it?

**Review by justinmc — APPROVED** [review]
LGTM 👍

**Comment by dannyvalentesonos** [comment]
@justinmc thanks for all the help! Everything is passing, and I don't have permissions to add labels. Will someone take care of merging this PR automatically? Thanks!

**Comment by justinmc** [comment]
Done!  This will be merged when the tree is green.  Thanks for fixing this.

**Pull request merged**